### DISCUSSION TITLE
How does one load several tags-Files?

### DISCUSSION THREAD

**Discussion by Sabancay** [universal-ctags/citre#156] [General]
I can"t figure out, how I can load the tags for a library every time I enter a project. In particular, I am experimenting with the Julia language. I use a global cache.

volker[535]~/.cache/tags$ ls -1
'!opt!volker!_julia!packages!.tags'
'!opt!volker!projects!julia!dataframes_training!.tags'
'!opt!volker!projects!julia!plots_training!.tags'

Whenever I work on a Julia-project, I want the tags from my Julia library '!opt!volker!_julia!packages!.tags' to be available in addition to the project tags. 
How would I configure citre to attain this?

**Comment by masatake** [comment]
I wonder https://github.com/universal-ctags/ctags/pull/3304#issuecomment-1528616715 and https://github.com/universal-ctags/ctags/issues/3704 are related to this topic. (readtags is a tool used in citre internally.)

**Reply by Sabancay** [reply]
Thanks for the quick reply. I found a use case for etags on https://github.com/JuliaEditorSupport/julia-ctags. I was hoping to adapt the idea to Citre. Would it be possible to configure the variable citre-tags-file-alist to attain such a goal? If yes, could you help with an example? 

  (require 'etags-table)

  (use-package julia-mode
    :ensure t
    :mode
    ("\\.jl\\'"  . julia-mode)
    :config
      (setq etags-table-alist
      '(
        (".*\\.jl$" "/home/volker/.julia/packages/TAGS")
        ))
      (setq tags-case-fold-search nil)

    ...
    )

**Comment by AmaiKinono** [comment]
It is possible for Citre to have such a feature, but that means to give up the sorter provided by readtags, and implement tags sorting in elisp, which is complicated, and runs slowly.

I would like to wait for https://github.com/universal-ctags/ctags/issues/3704 to be solved, so Citre could use it directly.

**Comment by swannodette** [comment]
In my case I noticed that I could achieve what I wanted by making a separate tags file for my dependencies (which don't change that often). This can be loaded by a separate tags backend and then it can be used as a fallback if the symbol can't be found in the default tags backend. Maybe this information will be useful to someone or perhaps this pattern can be made a bit more ergonomic?
